File: src/main/java/dev/jsinco/luma/commands/Subcommand.java
package dev.jsinco.luma.commands;

import dev.jsinco.luma.ThanksgivingEvent;
import org.bukkit.command.CommandSender;

import java.util.List;

public interface Subcommand {

    void execute(CommandSender sender, String[] args, ThanksgivingEvent plugin);

    List<String> tabComplete(CommandSender sender, String[] args, ThanksgivingEvent plugin);

    String permission();

    boolean isPlayerOnly();
}

File: src/main/java/dev/jsinco/lumacarnival/commands/SubCommand.kt
package dev.jsinco.lumacarnival.commands

import dev.jsinco.lumacarnival.CarnivalMain
import org.bukkit.command.CommandSender

interface SubCommand {

    fun execute(plugin: CarnivalMain, sender: CommandSender, args: Array<out String>)

    fun tabComplete(plugin: CarnivalMain, sender: CommandSender, args: Array<out String>): List<String>?

    fun permission(): String?

    fun playerOnly(): Boolean
}